File: frontend/src/api/jobs.ts
import type { Category, Country, JobSearchResponse } from '../types/job'

const API_BASE = import.meta.env.VITE_API_URL ?? 'http://localhost:8000/api'

async function apiFetch<T>(path: string, params: Record<string, string> = {}): Promise<T> {
  const qs = new URLSearchParams(params).toString()
  const res = await fetch(`${API_BASE}${path}${qs ? `?${qs}` : ''}`)
  if (!res.ok) {
    const body = await res.json().catch(() => ({}))
    throw new Error(body.detail ?? `Error ${res.status}`)
  }
  return res.json()
}

export function searchJobs(params: {
  country: string
  category?: string
  keyword?: string
  page?: number
}): Promise<JobSearchResponse> {
  const query: Record<string, string> = { country: params.country }
  if (params.category) query.category = params.category
  if (params.keyword) query.keyword = params.keyword
  if (params.page && params.page > 1) query.page = String(params.page)
  return apiFetch<JobSearchResponse>('/jobs', query)
}

export function getCountries(): Promise<Country[]> {
  return apiFetch<Country[]>('/countries')
}

export function getCategories(): Promise<Category[]> {
  return apiFetch<Category[]>('/categories')
}

File: backend/routers/jobs.py
from fastapi import APIRouter, HTTPException, Query
from models.job import Category, Country, JobSearchResponse
from services import adzuna

router = APIRouter()


@router.get("/jobs", response_model=JobSearchResponse)
async def get_jobs(
    country: str = Query(..., description="Código de país (ej: us, gb, mx)"),
    category: str = Query("", description="Categoría / rubro"),
    keyword: str = Query("", description="Palabra clave"),
    page: int = Query(1, ge=1, le=50),
):
    if country not in adzuna.SUPPORTED_COUNTRIES:
        raise HTTPException(
            status_code=400,
            detail=f"País '{country}' no soportado. Países válidos: {', '.join(adzuna.SUPPORTED_COUNTRIES)}",
        )
    return await adzuna.search_jobs(country, category, keyword, page)


@router.get("/countries", response_model=list[Country])
async def get_countries():
    return [Country(code=k, name=v) for k, v in adzuna.SUPPORTED_COUNTRIES.items()]


@router.get("/categories", response_model=list[Category])
async def get_categories():
    return [Category(code=k, name=v) for k, v in adzuna.CATEGORIES.items()]
